airflow/models.py

```diff
@@ -123,6 +123,7 @@ def clear_task_instances(tis, session, activate_dag_runs=True, dag=None):
     """
     job_ids = []
     for ti in tis:
```

Comment:
So this tracks the number of array jobs in each state? remind me how you determine the warning state?

Reply:
In here we just reset progress state in case that we call 'clear' on task (Through GUI).